### PR TITLE
Fix 404 search results when not on homepage

### LINKED ISSUE #278: Some results still lead to 404
Some results seem to be missing the `/doc` directory.

### PR DESCRIPTION
**Pull request by imaustink**

### CODE REVIEW

File: static/canjs.js [hunk 1]
```diff
@@ -95,7 +95,7 @@ function init() {
 				window.history.pushState(null, null, href);
 				navigate(href);
 			},
-			pathPrefix: window.pathPrefix,
+			pathPrefix: '/doc',
```
Comment (imaustink) [inline static/canjs.js]: Hard setting this to the correct value for the live site means changing the testing environment to match. I am open to other ideas, but this is the cleanest option I found.

**Comment by imaustink** [comment]
@chasenlehara any idea why the build is failing?

**Comment by chasenlehara** [comment]
This line probably needs to be changed as well: https://github.com/canjs/bit-docs-html-canjs/blob/032ecbc25ae6528d1325843d5a3271a70157b9b4/test/search.js#L21

File: static/search.js [hunk 2]
```diff
@@ -514,16 +514,15 @@ var Search = Control.extend({
...
 					docUrl: function(){
-						if(!self.options.pathPrefix){
-							return this.url;
+						if(self.options.pathPrefix){
+							return self.options.pathPrefix + "/" + this.url;
 						}
-
-						var root = joinURIs(window.location.href, self.options.pathPrefix);
-						if(root.substr(-1) === "/"){
-							root = root.substr(0, root.length-1);
```
Comment (chasenlehara) [inline static/search.js]: None of the `root` logic was necessary?

**Comment by imaustink** [comment]
@chasenlehara you were right! Not sure how I missed that. I thought I had tests passing locally but I was mistaken.

File: static/search.js (hunk 2 above)
Comment (imaustink) [inline static/search.js]: As far as I can tell it was part of the problem and the new setup forgoes the need to have it at all.

**Pull request merged**